### PR TITLE
add a callback function to display the selected wells

### PR DESCRIPTION
**Pull request by VincentNevermore**
image

### CODE REVIEW

File: react/src/lib/components/DeckGLMap/layers/drawing/drawingLayer.tsx [hunk 1]
```diff
@@ -32,6 +32,7 @@ class CustomModifyMode extends ModifyMode {
         super.handleKeyUp(event, props);
 
         if (event.key === "Delete") {
+            console.log("In");
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/drawing/drawingLayer.tsx:35]: Please clean up loggging.

**Review by hkfb — CHANGES_REQUESTED** [review]

File: react/src/lib/components/DeckGLMap/layers/LassoLayer/lassoLayer.stories.tsx [hunk 2]
```diff
@@ -123,3 +123,51 @@ const enableLassoArgs = {
...
+        ...disableLassoArgs,
+        layers: [
+            {
+                "@@type": "WellsLayer",
+                data: "@@#resources.wellsData",
+            },
+            {
+                "@@type": "LassoLayer",
+                visible: true,
+                data: "@@#resources.wellsData",
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/LassoLayer/lassoLayer.stories.tsx:152]: I hadn't noticed this before; is it necessary for the selection layer to have a copy of the data?

Comment (VincentNevermore) [reply react/src/lib/components/DeckGLMap/layers/LassoLayer/lassoLayer.stories.tsx:152]: I think now we can delete it for now at least.

File: react/src/lib/components/DeckGLMap/layers/LassoLayer/lassoLayer.tsx [hunk 3]
```diff
@@ -16,6 +16,8 @@ export interface LassoLayerProps<D> extends ExtendedLayerProps<D> {
     lineWidthScale: number;
     lineStyle: LineStyleAccessor;
     wellHeadStyle: WellHeadStyleAccessor;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    getSelectedWellsData: (pickingInfos: any[]) => void;
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/LassoLayer/lassoLayer.tsx:20]: Suggest having a more generic prop, eg. `handleSelection` to cover any selected data.

File: react/src/lib/components/DeckGLMap/layers/LassoLayer/lassoLayer.stories.tsx [hunk 4]
```diff
@@ -119,7 +118,53 @@ const enableLassoArgs = {
...
+                .map((item) => item.properties["name"]) as string[];
+            setData(selectedWells);
+        },
+        []
+    );
+    const lassoArgsWithSelectedWellsDataCallback: Record<string, unknown> = {
+        ...disableLassoArgs,
+        layers: [
+            {
+                "@@type": "WellsLayer",
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/LassoLayer/lassoLayer.stories.tsx:144]: Is it necessary to use the JSON format for specifying layers here? Could you use the typed classes, ie. `[new WellsLayer({data: ...}), new LassoLayer({...})]`?

**Review by hkfb — CHANGES_REQUESTED** [review]

File: react/src/lib/components/DeckGLMap/layers/LassoLayer/lassoLayer.stories.tsx (hunk 4 above)
Comment (VincentNevermore) [reply react/src/lib/components/DeckGLMap/layers/LassoLayer/lassoLayer.stories.tsx:144]: hmmm I updated and it seems it does not like the typed classes format..
image

Comment (hkfb) [reply react/src/lib/components/DeckGLMap/layers/LassoLayer/lassoLayer.stories.tsx:144]: Ok, never mind then. Suggest that we fix this in the future.

File: react/src/lib/components/DeckGLMap/layers/LassoLayer/lassoLayer.stories.tsx [hunk 5]
```diff
@@ -119,7 +118,53 @@ const enableLassoArgs = {
...
+                data: "@@#resources.wellsData",
+            },
+            {
+                "@@type": "LassoLayer",
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/LassoLayer/lassoLayer.stories.tsx:148]: Suggest that we rename to "BoxSelectionLayer".

**Review by hkfb — CHANGES_REQUESTED** [review]

File: react/src/lib/components/DeckGLMap/layers/layersDefaultProps.ts [hunk 6]
```diff
@@ -158,10 +158,10 @@ export const layersDefaultProps: Record<string, unknown> = {
...
-    LassoLayer: {
-        "@@type": "LassoLayer",
-        name: "Lasso",
-        id: "lasso-layer",
+    BoxSelectionLayer: {
+        "@@type": "BoxSelectionLayer",
+        name: "boxSelection",
+        id: "boxSelection-layer",
         pickable: true,
         visible: false,
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/layersDefaultProps.ts:166]: The layer should be visible by default.

**Review by hkfb — CHANGES_REQUESTED** [review]

File: react/src/lib/components/DeckGLMap/layers/BoxSelectionLayer/boxSelectionLayer.stories.tsx [hunk 7]
```diff
@@ -22,20 +22,20 @@ const useStyles = makeStyles({
...
+export const boxSelection: ComponentStory<typeof DeckGLMap> = (args) => {
     const [editedData, setEditedData] = React.useState(args.editedData);
     const [argsState, setArgsState] =
         React.useState<Record<string, unknown>>(enableLassoArgs);
     const [state, setState] = React.useState<boolean>(true);
 
     const handleChange = React.useCallback(() => {
-        const lassoLayer = enableLassoArgs.layers.filter(
-            (item) => item["@@type"] === "LassoLayer"
+        const BoxSelectionLayer = enableLassoArgs.layers.filter(
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/BoxSelectionLayer/boxSelectionLayer.stories.tsx:32]: Convention is to not capitalize the first letter.

**Review by hkfb — CHANGES_REQUESTED** [review]

**Review by hkfb — CHANGES_REQUESTED** [review]

File: react/src/lib/components/DeckGLMap/layers/BoxSelectionLayer/boxSelectionLayer.tsx [hunk 8]
```diff
@@ -16,6 +16,8 @@ export interface LassoLayerProps<D> extends ExtendedLayerProps<D> {
     lineWidthScale: number;
     lineStyle: LineStyleAccessor;
     wellHeadStyle: WellHeadStyleAccessor;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    handleSelection: (pickingInfos: any[]) => void;
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/BoxSelectionLayer/boxSelectionLayer.tsx:20]: Please fix this lint condition. Try the PickInfo type.

File: react/src/lib/components/DeckGLMap/layers/BoxSelectionLayer/boxSelectionLayer.stories.tsx [hunk 9]
```diff
@@ -117,9 +117,54 @@ const enableLassoArgs = {
...
+            },
+        ],
+    };
+    return (
+        <>
+            <div className={useStyles().main}>
+                <DeckGLMap
+                    id={"DeckGL-Map"}
+                    {...lassoArgsWithSelectedWellsDataCallback}
+                    legend={{ visible: false }}
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/BoxSelectionLayer/boxSelectionLayer.stories.tsx:160]: The legend prop has been deprecated.

File: react/src/lib/components/DeckGLMap/layers/BoxSelectionLayer/boxSelectionLayer.stories.tsx [hunk 10]
```diff
@@ -22,20 +23,20 @@ const useStyles = makeStyles({
     },
 });
 
-export const lassoSelection: ComponentStory<typeof DeckGLMap> = (args) => {
+export const boxSelection: ComponentStory<typeof DeckGLMap> = (args) => {
     const [editedData, setEditedData] = React.useState(args.editedData);
```
Comment (hkfb) [inline react/src/lib/components/DeckGLMap/layers/BoxSelectionLayer/boxSelectionLayer.stories.tsx:27]: What is this used for? I think it can be removed.

**Review by hkfb — CHANGES_REQUESTED** [review]

File: react/src/lib/components/DeckGLMap/layers/BoxSelectionLayer/boxSelectionLayer.stories.tsx (hunk 9 above)
Comment (hkfb) [reply react/src/lib/components/DeckGLMap/layers/BoxSelectionLayer/boxSelectionLayer.stories.tsx:147]: The `legend` prop has been deprecated.

**Comment by hkfb** [comment]
Please attach any issue ids that this PR solves.

**Pull request merged**